third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1726,83 +1817,60 @@ impl Frame {
                 let field_mut_ref_ty = ty_builder.create_ref_ty(field_ty, true)?;
                 interpreter.operand_stack.push_ty(field_mut_ref_ty)?;
             },
+            Bytecode::ImmBorrowVariantField(fh_idx) | Bytecode::MutBorrowVariantField(fh_idx) => {
+                let is_mut = matches!(instruction, Bytecode::MutBorrowVariantField(..));
+                let field_info = resolver.variant_field_info_at(*fh_idx);
+                let ty = interpreter.operand_stack.pop_ty()?;
+                let expected_ty = resolver.create_struct_type(&field_info.definition_struct_type);
```

Comment:
nit: you can use `ty_builder` directly? All these calls to loader/resolver are very annoying, given that we make them just because the code is placed there...

Reply:
Same for other places